### ISSUE TITLE
Delayed invalidationBehavior issue with long running queries

### ISSUE THREAD

**Issue by MateuszGroth** [reduxjs/redux-toolkit#4347]
### Issue description

Whenever there is a pending query on some endpoint 'A', invalidation of cache (re-fetch when subscribed) on endpoint 'B' is only triggered when the endpoint 'A' query finishes. 

Consider a scenario where you have two queries getPosts and getMyPosts. Let's assume that getPosts takes around 25 seconds to finish and getMyPosts returns data immediately (200ms). 

Steps to reproduce:
- Subscribe to getPosts and getMyPosts
- invalidate getMyPosts after a couple of seconds (white getPosts query is still pending)

Expected result:
- my posts data is re-fetched right after the invalidation (delayed as a task/microTask)

Actual result:
- my posts data is re-fetched only when the getPosts query finishes

**Comment by EskiMojo14** [comment]
yeah, this is a [2.0 change](https://redux-toolkit.js.org/usage/migrating-rtk-2#rtk-query-behaviour-changes) - you can set `invalidationBehavior: 'immediate'` to use the 1.9 behaviour.

**Comment by MateuszGroth** [comment]
@EskiMojo14 I get the idea behind the 'delayed' option as I also experienced issues with the 'immediate'. However, I feel like this is delayed a bit too much if you know what I mean. Why would the invalidation of a query wait for another query to be finished loading?

**Comment by markerikson** [comment]
I think there was some discussion in the original issue around possibly trying to make the behavior more targeted in _which_ queries it waits for, but we opted to defer that aspect in order to get 2.0 out the door.

**Comment by BrenoVelasco** [comment]
Do we plan on adding support to an "ignore" feature? There are some long-running queries that we poll every X seconds and we'd love to be able to `skipDelayInvalidation` or something for these. Thanks!

**Comment by elburro1887** [comment]
I would also support an "ignore" feature, or a revert to the default value. 

I just discovered that this is the reason why my caches were only sporadically updating, and I have been working with RTK Query for over a year. 

I have a periodic query, which somehow was screwing up the cache updates because of this setting. I changed it to 'immediate' now, and everything works flawlessly, finally!

This flag is not documented in the Cache Docs (https://redux-toolkit.js.org/rtk-query/usage/cache-behavior) and I'm sure I'm not the only one that go tripped up by it, since there are a lot of questions and confusion about the cache behaviour on Stack Overflow and Reddit, etc..

Therefore I would argue that the default here is not "smart" default, so I would recommend to change it back.

**Comment by markerikson** [comment]
@elburro1887 it's something we can consider, but changing that would have to wait until the next major.

Do you have any links to these other discussions you're referencing?  I haven't been looking at SO much lately, but I don't remember seeing any relevant discussions on /r/reactjs or /r/reduxjs .

**Comment by elburro1887** [comment]
Thanks for the fast reply. 
To be fair, I don't have any concrete links on hand, I just have my own anecdotal experience.
I was often looking for solutions online and (until now) never found anything concrete that would solve my issues, and from my feeling when reading lots of online material a lot of people seem to struggle with "unexpected" cache behavior.
